### PR TITLE
Integration of RF model to predict efficacy of immune checkpoint blockade (BioModels)

### PR DESCRIPTION
**Pull request by anuprulez**
The PR adds a tool that uses an RF model, downloaded via a Docker container, to predict the efficacy of immune checkpoint blockage on the provided test data. The RF model is available via BioModels (https://www.ebi.ac.uk/biomodels/BIOMD0000001066). The Docker container available in BioModels (https://www.ebi.ac.uk/biomodels/BIOMD0000001066#Files) for prediction has been customised to reduce its uncompressed size (by half, from over 4 GBs to less than 2). The newly developed container is linked in the tool (docker.io/anupkumar/immune-checkpoint:latest) and also stored on Docker Hub (https://hub.docker.com/repository/docker/anupkumar/biomd0000001066/general). The script used to create this customised container is available on GitHub (https://github.com/anuprulez/biomodelsml-docker/tree/master/rf_immune_checkpoints_blockage).

test file(s): 
[test_data.xlsx](https://github.com/bgruening/galaxytools/files/13281243/test_data.xlsx)
[test_data.csv](https://github.com/bgruening/galaxytools/files/13281244/test_data.csv)

Use one of these files to test the tool on Galaxy server.

Tool's UI (on a local Galaxy instance):

local_instance

ping @bgruening

### CODE REVIEW

File: tools/biomodelsML/bio_ml_tool.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<tool id="bio_ml_tool" name="Random Forest model" version="0.0.1" profile="22.05">
+    <description>to predict efficacy of immune checkpoint blockade across multiple cancer patient cohorts</description>
+    <requirements>
+        <container type="docker">docker.io/anupkumar/immune-checkpoint:latest</container>
+    </requirements>
+    <command><![CDATA[
+    cp /home/\$NB_USER/forest16.onnx `pwd` && python '$biom_script'
```
Comment (bgruening) [inline tools/biomodelsML/bio_ml_tool.xml:7]:
```suggestion
    cp /home/\$NB_USER/forest16.onnx ./ &&
    python '$biom_script'
```

File: tools/biomodelsML/bio_ml_tool.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <data format="tabular" name="output_file" label="Predicted data"></data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_file" value="test_data.tabular" ftype="tabular" />
+            <output name="output_file" file="pred_data.tabular" compare="sim_size" delta="5" />
+        </test>
+        <test>
+            <param name="input_file" value="test_data.csv" ftype="csv" />
+            <output name="output_file" file="pred_data.tabular" compare="sim_size" delta="5" />
```
Comment (bgruening) [inline tools/biomodelsML/bio_ml_tool.xml:52]: sim_size is not recommendade, can we use assert_contents here, number of columns, number of rows https://docs.galaxyproject.org/en/latest/dev/schema.html#id112

File: tools/biomodelsML/bio_ml_tool.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<tool id="bio_ml_tool" name="Random Forest model" version="0.0.1" profile="22.05">
```
Comment (bgruening) [inline tools/biomodelsML/bio_ml_tool.xml:1]:
```suggestion
<tool id="biomodel_BIOMD0000001066" name="Random Forest model" version="1" profile="22.05">
```

I have taken those from https://www.ebi.ac.uk/biomodels/BIOMD0000001066#History

File: tools/biomodelsML/bio_ml_tool.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    <description>to predict efficacy of immune checkpoint blockade across multiple cancer patient cohorts</description>
+    <requirements>
+        <container type="docker">docker.io/anupkumar/immune-checkpoint:latest</container>
```
Comment (bgruening) [inline tools/biomodelsML/bio_ml_tool.xml:4]: maybe also rename the container to fit the tool_id?

Comment (anuprulez) [reply tools/biomodelsML/bio_ml_tool.xml]: thanks, fixed!

File: tools/biomodelsML/bio_ml_tool.xml (hunk 3 above)
Comment (anuprulez) [reply tools/biomodelsML/bio_ml_tool.xml]: sounds good, fixed it.

File: tools/biomodelsML/bio_ml_tool.xml (hunk 2 above)
Comment (anuprulez) [reply tools/biomodelsML/bio_ml_tool.xml]: yes, sound good!

File: tools/biomodelsML/bio_ml_tool.xml (hunk 1 above)
Comment (anuprulez) [reply tools/biomodelsML/bio_ml_tool.xml]: I fixed the permissions in the container, let's see. Not sure if we need to create a new user just like the GPU based container.

**Review by bgruening — APPROVED** [review]

**Review by bgruening — APPROVED** [review]

**Pull request merged**